### DISCUSSION TITLE
Possibility to load old Pybricks versions

### DISCUSSION THREAD

**Discussion by Marlow925** [pybricks/support#1150] [Ideas]
Hi,

would it be possible to load the UI for older PyBricks releases (1.1.0 specificly) to be able to use and upload legacy scripts ?

There are some use cases, where scripts exist made for 1.1.0 / firmware 3.1.0.

I have managed to upload the older firmware, but it then complains about the mpy file format being wrong. The script, I'm trying to upload was abandoned, when firmware 3.2.0 was released, as the changes were too complicated to implement.

**Comment by laurensvalk** [comment]
Can you elaborate on the use case? What does the code look like?

There shouldn't be much breaking changes going from 3.1 to 3.2. What doesn't work anymore?

It's technically possible to go back since everything is open source, but this is not something we actively encourage or support.

**Comment by Marlow925** [comment]
This script by Lok24: [[https://drive.google.com/file/d/1r2vjC0xOvaD5RqUxi-aml75tHumsEmvb/view?usp=share_link](url)](motorcontrol.py) allows multiple motors using the PU remote, disabling the watchdog, so trains don't stop, when they drive out of reach of the remote (large layouts need that), smooth accelleration and autodetects if you're using train motors or technic motors.

It was originally published here: https://www.eurobricks.com/forum/index.php?%2Fforums%2Ftopic%2F187081-control-your-trains-without-smart-device-with-pybricks%2F

The developer of the script stated: "It took too much effort to me to adapt the programs and description to the changes pybricks made". Referring to the massive changes from firmware 3.1 to 3.2.

Trainmotors still work, when using firmware 3.2 and beyond. Technic motors aren't detected nor working and that was actually the greatest feature of this script.

I have in the meanwhile found a way to load the script with firmware 3.1.0. It's just cumbersome.

/M

**Reply by laurensvalk** [reply]
All of that should still work, so maybe we can make a nice program that suits your needs.

We can share the new program so that others can benefit from it too. In the long run, this is more convenient for everyone compared to trying to run old software. 

Can you describe in words what you would like your program to do? We'd be happy to put together some examples.

**Comment by Marlow925** [comment]
We have already (within our community) started to fix the issues in that script.

They actually turned out to be fairly minor. I figure, the original developer just lost the will to maintain the scripts over more major issues in some of his other scripts.

It turns out, that firmware 3.1.0 actually poses other problems (like the remote won't reconnect, if it disconnected. That works in firmware 3.2.0)

/M
